### PR TITLE
FieldDisplay: updated condition for No data text display

### LINKED ISSUE #33599: Hide "No Data" text
**What would you like to be added**:
I want my panel to be empty if there is no data. 
So I put a space in the `No Value: What to show when there is no value` field.
This space is being ignored, any other character will work. 

**Why is this needed**:
By default this field is empty and yet `No Data` is displayed.
The field and the output should match. ie: the field should also say: `No Data`, then it would actually match.

This may be a non-issue for others, but please provide a solution on how to make this `No Data` disappear.

Screenshot 2021-05-02 at 07 11 20

### PR DESCRIPTION
**Pull request by PraveenJP**
**What this PR does / why we need it:**
In this fix, If the response and the 'No value' field is empty, the panel will show empty.

**Which issue(s) this PR fixes:**
Fixes #33599

**Special notes for your reviewer**:
I have updated the 'fieldDisplay.test.js' based on the above changes

### CODE REVIEW

File: packages/grafana-data/src/field/fieldDisplay.test.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ describe('FieldDisplay', () => {
   it('Should return field with default text when no mapping or data available', () => {
     const options = createEmptyDisplayOptions();
     const display = getFieldDisplayValues(options);
-    expect(display[0].display.text).toEqual('No data');
+    expect(display[0].display.text).toEqual('');
     expect(display[0].display.numeric).toEqual(0);
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldDisplay.test.ts:106]: i think for your scenario you need to also add field config with displayName set to empty string

Comment (PraveenJP) [reply packages/grafana-data/src/field/fieldDisplay.test.ts]: I have added new test case 'Should return empty when no field mapped and no data'

**Comment by PraveenJP** [comment]
Could anyone please review this changes?

**Review by torkelo — CHANGES_REQUESTED** [review]
This is breaking the default behavior for "no data" handling

**Review by hugohaggmark — COMMENTED** [review]
Any update to this @PraveenJP

**Comment by PraveenJP** [comment]
> This is breaking the default behavior for "no data" handling

Can you please clarify which scenario is breaking? I am a new bee can you please guide me on this?

**Comment by PraveenJP** [comment]
**Sample Output:**

**Here no value field not updated with empty data**:
2021-06-10_13h25_55

**Here no value field updated with empty data**:
image

I hope this is the expectation right?

**Comment by pkolyvas** [comment]
> > This is breaking the default behavior for "no data" handling
> 
> Can you please clarify which scenario is breaking? I am a new bee can you please guide me on this?

I believe the response is that the current default behavior must be maintained as it's a pattern that's common and understood across Grafana. The PR should at least maintain the current "No Data" display without any additional user input or configuration. A blank panel is less desirable as it looks like a display error.

**Pull request closed without merging**